packages/fx-core/src/plugins/solution/fx-solution/commonQuestions.ts

```diff
@@ -671,24 +691,68 @@ async function createNewResourceGroup(
   return ok(response.name);
 }
 
-async function checkResourceGroupExistence(
+function handleRestError<T>(
+  restError: RestError,
+  resourceGroupName: string,
+  subscriptionId: string,
+  subscriptionName: string
+): Err<T, FxError> {
+  const subscriptionInfoString =
+    subscriptionId + (subscriptionName.length > 0 ? `(${subscriptionName})` : "");
+  if (restError.statusCode === 403) {
+    return err(
+      returnUserError(
+        new Error(
+          `Unauthorized to check the existence of resource group '${resourceGroupName}' in subscription '${subscriptionInfoString}'. Please check your Azure subscription.`
+        ),
+        SolutionSource,
+        SolutionError.Unauthorized
+      )
+    );
+  } else {
+    // Avoid sensitive information like request headers in the error message.
+    const rawErrorString = JSON.stringify({
+      code: restError.code,
+      statusCode: restError.statusCode,
+      body: restError.body,
+      name: restError.name,
+      message: restError.message,
+    });
+    return err(
+      returnUserError(
+        new Error(
+          `Failed to check the existence of resource group '${resourceGroupName}' in subscription '${subscriptionInfoString}', raw error: '${rawErrorString}'`
+        ),
+        SolutionSource,
+        SolutionError.FailedToCheckResourceGroupExistence
+      )
+    );
+  }
+}
+
+export async function checkResourceGroupExistence(
   rmClient: ResourceManagementClient,
-  resourceGroupName: string
+  resourceGroupName: string,
+  subscriptionId: string,
+  subscriptionName: string
 ): Promise<Result<boolean, FxError>> {
   try {
     const checkRes = await rmClient.resourceGroups.checkExistence(resourceGroupName);
     return ok(!!checkRes.body);
   } catch (e) {
-    // The `checkExistence()` call returns an error with empty error message in some conditions.
-    // Wrap the error to prevent showing empty error message to the users.
     let error;
-    const errorMessage = `Failed to check the existence of the resource group '${resourceGroupName}', error: '${e}'`;
-    if (e instanceof Error) {
+    if (e instanceof RestError) {
+      return handleRestError(e, resourceGroupName, subscriptionId, subscriptionName);
+    } else if (e instanceof Error) {
       // reuse the original error object to prevent losing the stack info
-      e.message = errorMessage;
+      e.message = `Failed to check the existence of resource group '${resourceGroupName}', error: '${e}'`;
```

Comment:
> `error: '${e}'`
do we need to stringify the object as well here?

Reply:
The reason here is for some other types of non-RestError (like network error), the user can know:
1. The error is related to checking resource group
2. The true reason of the error so he can take action or report issue.

So I wrap these two pieces of information into this error message.